packages/dds/tree/src/simple-tree/api/configuration.ts

```diff
@@ -345,15 +372,55 @@ export function checkUnion(
 		);
 	}
 
+	if (records.length > 1) {
+		ambiguityErrors.push(
+			`More than one kind of record allowed within union (${formatTypes(records)}). This would require type disambiguation which is not supported by records during import or export.`,
+		);
+	}
+
 	if (maps.length > 0 && arrays.length > 0) {
 		ambiguityErrors.push(
 			`Both a map and an array allowed within union (${formatTypes([...arrays, ...maps])}). Both can be implicitly constructed from iterables like arrays, which are ambiguous when the array is empty.`,
 		);
 	}
 
-	if (objects.length > 0 && maps.length > 0) {
+	/**
+	 * Constructs a sentence-formatted list of entries, using commas and "and" as appropriate.
+	 */
+	function sentenceList(entries: readonly string[]): string {
+		switch (entries.length) {
+			case 0: {
+				return "";
+			}
+			case 1: {
+				return entries[0] ?? oob();
```

Comment:
you never use either of these. I think the default case should handle all the cases this code actually needs which makes this utility probably not worth having.

I also can't Imagin a use case where the 0 item case would be grammatically well formed anyway. For example, `Only ${list} are allowed here` would not be able to use this helper in the 0 case.